ldapauthenticator/ldapauthenticator.py

```diff
@@ -518,8 +519,11 @@ async def authenticate(self, handler, data):
                     attributes=self.group_attributes,
                 )
                 if found:
-                    break
-            if not found:
+                    ldap_groups.append(group)
+                    # we currently only use this in check_allowed,
+                    # so we could stop here, as only one match is relevant
+                    # break
```

Comment:
I'm considering `manage_groups` etc, and if that is set, we want all groups - but what we are checking for isn't all groups or similar, just a subset of groups the user is part of among the `allowed_groups`.

Due to that, the list in `ldap_groups` will be incomplete no matter what, so I'm absolutely open to breaking here instead of doing additional queries.